jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -137,6 +137,25 @@ private int get(Content content, byte[] bytes, int offset, int length)
         return consumed;
     }
 
+    private int get(Content content, ByteBuffer des)
+    {
+        var capacity = des.remaining();
+        var src = content.getByteBuffer();
+        if (src.remaining() > capacity)
+        {
+            var copy = src.duplicate().limit(src.position() + capacity);
+            des.put(copy);
+            src.position(src.position() + (capacity - copy.remaining()));
```

Comment:
There is no need to allocate via `duplicate()`, this should work:

```suggestion
            int limit = src.limit();
            src.limit(src.position() + capacity);
            des.put(src);
            src.limit(limit);
```

Reply:
It's more efficient, thanks!